### PR TITLE
Dispose overlay entries

### PR DESCRIPTION
**Pull request by tgucio**
As `OverlayEntry` implements a `Listenable` that is e.g. subscribed to by the `Navigator`, it should be disposed when no longer used. This PR adds the missing `dispose()` calls in all the places I could find they were missing.

Test-exempt: adding missing dispose() methods without other functional changes

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

**Comment by navaronbracke** [comment]
Would this unblock removing the leak tracking config overrides on some `testWidgetsWithLeakTracking()` usages?

This PR does not modify any tests, hence why I'm asking.

File: packages/flutter/lib/src/widgets/magnifier.dart [hunk 1]
```diff
@@ -257,7 +256,7 @@ class MagnifierController {
       to: Navigator.maybeOf(context)?.context,
     );
 
-   _overlayEntry = OverlayEntry(
+    _overlayEntry = OverlayEntry(
```
Comment (justinmc) [inline packages/flutter/lib/src/widgets/magnifier.dart:259]: Good catch on the space 👾

**Review by justinmc — APPROVED** [review]
LGTM 👍 

Thanks for cleaning this up!

**Comment by justinmc** [comment]
@polina-c Is there a way to test these changes using your leak tracking tests?  Or should it be test exempt?

**Comment by polina-c** [comment]
OverlayEntry is not instrumented to be tracked for leaks yet. So, not yet. Added it to list of classes to instrument.

**Comment by Hixie** [comment]
test-exempt: will be tracked

is there a bug# to follow that work?

**Pull request merged**